sonar-application/src/main/java/org/sonar/application/ClusterProperties.java

```diff
@@ -0,0 +1,162 @@
+
+/**
+ * Properties of the cluster configuration
+ */
+final class ClusterProperties {
+
+  enum Parameter {
+    ENABLED("sonar.cluster.enabled", Boolean.FALSE.toString()),
+    MEMBERS("sonar.cluster.members", ""),
+    PORT("sonar.cluster.port", Integer.toString(9003)),
+    PORT_AUTOINCREMENT("sonar.cluster.port_autoincrement", Boolean.FALSE.toString()),
+    INTERFACES("sonar.cluster.interfaces", ""),
+    NAME("sonar.cluster.name", ""),
+    LOG_LEVEL("sonar.cluster.log.level", "WARN");
+
+    private String name;
+    private String defaultValue;
+
+    Parameter(String name, String defaultValue) {
+      this.name = name;
+      this.defaultValue = defaultValue;
+    }
+
+    String getName() {
+      return name;
+    }
+
+    String getDefaultValue() {
+      return defaultValue;
+    }
+
+    boolean getDefaultValueAsBoolean() {
+      return "true".equalsIgnoreCase(defaultValue);
+    }
+
+    Integer getDefaultValueAsInt() {
+      if (defaultValue != null && !"".equals(defaultValue)) {
+        try {
+          return Integer.parseInt(defaultValue);
+        } catch (NumberFormatException e) {
+          throw new IllegalStateException("Default value of property " + name + " is not an integer: " + defaultValue, e);
+        }
+      }
+      return null;
+    }
+  }
+
+  private int port;
+  private boolean enabled;
+
+  private boolean portAutoincrement;
+  private String members;
+  private String interfaces;
+  private String name;
+  private String logLevel;
+
+  ClusterProperties(Props props) {
+    port = props.valueAsInt(Parameter.PORT.getName(), Parameter.PORT.getDefaultValueAsInt());
+    enabled = props.valueAsBoolean(Parameter.ENABLED.getName(), Parameter.ENABLED.getDefaultValueAsBoolean());
+    portAutoincrement = props.valueAsBoolean(Parameter.PORT_AUTOINCREMENT.getName(), Parameter.PORT_AUTOINCREMENT.getDefaultValueAsBoolean());
+    interfaces = props.value(Parameter.INTERFACES.getName(), Parameter.INTERFACES.getDefaultValue());
+    name = props.value(Parameter.NAME.getName(), Parameter.NAME.getDefaultValue());
+    logLevel = props.value(Parameter.LOG_LEVEL.getName(), Parameter.LOG_LEVEL.getDefaultValue());
+    members = props.value(Parameter.MEMBERS.getName(), Parameter.MEMBERS.getDefaultValue());
+  }
+
+  void populateProps(Props props) {
+    props.set(Parameter.PORT.getName(), Integer.toString(port));
+    props.set(Parameter.ENABLED.getName(), Boolean.toString(enabled));
+    props.set(Parameter.PORT_AUTOINCREMENT.getName(), Boolean.toString(portAutoincrement));
+    props.set(Parameter.INTERFACES.getName(), interfaces);
+    props.set(Parameter.NAME.getName(), name);
+    props.set(Parameter.LOG_LEVEL.getName(), logLevel);
+    props.set(Parameter.MEMBERS.getName(), members);
+  }
+
+  int getPort() {
+    return port;
+  }
+
+  boolean isEnabled() {
+    return enabled;
+  }
+
+  boolean isPortAutoincrement() {
+    return portAutoincrement;
+  }
+
+  String getMembers() {
```

Comment:
value of property is not used as is, we already split it to add port when missing
so, my comment stands